app/views/components/AnswerSection.scala.html

```diff
@@ -65,20 +62,32 @@ <h2 class="govuk-heading-m">@title</h2>
 }
 
 @content = {
-    <dl class="govuk-summary-list">
-        @govukSummaryList(
-            SummaryList(
-                rows = section.rows
+    @if(section.rows.nonEmpty) {
+        <dl class="govuk-summary-list">
+            @govukSummaryList(
+                SummaryList(
+                    rows = section.rows
+                )
             )
-        )
-    </dl>
+        </dl>
+    }
 
-    @section.children.map(this(_, nestingLevel + 1))
+    @if(section.children.nonEmpty) {
+        @section.children.map(this(_, nestingLevel + 1))
+    }
 
     @section.viewLink.map { link =>
         <p class="govuk-body">
-            <a class="govuk-link" href=@link.href id="view_link.id">
-                @messages("summaryDetails.link") <span class="govuk-visually-hidden">@link.visuallyHidden</span>
+            <a class="govuk-link" href=@link.href id="@link.id">
+                @messages(link.text) <span class="govuk-visually-hidden">@link.visuallyHidden</span>
+            </a>
+        </p>
+    }
+
+    @section.secondViewLink.map { link =>
+        <p class="govuk-body">
+            <a class="govuk-link" href=@link.href id="@link.id">
+                @messages(link.text) <span class="govuk-visually-hidden">@link.visuallyHidden</span>
```

Comment:
Rather than having `viewLink` and `secondViewLink` could we have `viewLinks: Seq[Link]`?

Reply:
Good shout!